mPower2/mPower2/StudyBurstScheduleManager.swift

```diff
@@ -616,4 +616,10 @@ extension Array {
             last -= 1
         }
     }
+    
+    public func randomFirst() -> Element? {
+        guard self.count > 1 else { return self.first }
+        let rand = Int(arc4random_uniform(UInt32(self.count)))
```

Comment:
all this apparently gets much easier in Swift 4.2:

return self.randomElement()

Unfortunately that requires Xcode 10. :P